src/libpmem2/source_posix.c

```diff
@@ -174,3 +174,14 @@ pmem2_source_alignment(const struct pmem2_source *src, size_t *alignment)
 
 	return 0;
 }
+
+/*
+ * pmem2_source_get_fd -- get file descriptor from provided source
+ */
+int
+pmem2_source_get_fd(const struct pmem2_source *src)
+{
+	LOG(3, "src %p", src);
+
+	return src->value.fd;
```

Comment:
This is only valid for `src->type == PMEM2_SOURCE_FD`.

I think we should change this to `int pmem2_source_get_fd(const struct pmem2_source *src, int *fd)`, and return `PMEM2_E_FILE_HANDLE_NOT_SET` (windows) or `PMEM2_E_FILE_DESCRIPTOR_NOT_SET` (linux) when there's no handle/fd to return.